libs/api/domains/social-insurance/src/lib/dtos/pensionCalculation.input.ts

```diff
@@ -0,0 +1,112 @@
+import { InputType, Field, registerEnumType } from '@nestjs/graphql'
+
+export enum BasePensionType {
+  Retirement = 'Retirement',
+  FishermanRetirement = 'FishermanRetirement',
+  Disability = 'Disability',
+  Rehabilitation = 'Rehabilitation',
+  HalfRetirement = 'HalfRetirement',
+}
+
+registerEnumType(BasePensionType, {
+  name: 'SocialInsurancePensionCalculationBasePensionType',
+})
+
+export enum LivingCondition {
+  LivesAlone = 'LivesAlone',
+  DoesNotLiveAlone = 'DoesNotLiveAlone',
+}
+
+registerEnumType(LivingCondition, {
+  name: 'SocialInsurancePensionCalculationLivingCondition',
+})
+
+export enum PensionStart {
+  Starts2017OrBefore = 'Starts2017OrBefore',
+  Starts2018OrLater = 'Starts2018OrLater',
+}
+
+registerEnumType(PensionStart, {
+  name: 'SocialInsurancePensionCalculationPensionStart',
+})
+
+export enum PeriodIncomeType {
+  Month = 'Month',
+  Year = 'Year',
+}
+
+registerEnumType(PeriodIncomeType, {
+  name: 'SocialInsurancePensionCalculationPeriodIncomeType',
+})
+
+@InputType('SocialInsurancePensionCalculationInput')
+export class PensionCalculationInput {
+  @Field(() => BasePensionType, { nullable: true })
+  typeOfBasePension?: BasePensionType
+
+  @Field(() => Boolean, { nullable: true })
+  hasSpouse?: boolean
+
+  @Field(() => LivingCondition, { nullable: true })
+  livingCondition?: LivingCondition
+
+  @Field(() => Boolean, { nullable: true })
+  mobilityImpairment?: boolean
+
+  @Field(() => String, { nullable: true })
+  startDate?: string
+
+  @Field(() => Number, { nullable: true })
+  childCount?: number
```

Comment:
FYI you don't need to explicitly declare the Boolean or the String. However, you need to assign the number (`childCount`) to either Int or Float, according to the[ NestJS docs](https://docs.nestjs.com/graphql/resolvers)

`The type function is required when there's the potential for ambiguity between the TypeScript type system and the GraphQL type system. Specifically: it is not required for string and boolean types; it is required for number (which must be mapped to either a GraphQL Int or Float).`